scylla/src/transport/downgrading_consistency_retry_policy.rs

```diff
@@ -171,7 +171,18 @@ impl RetrySession for DowngradingConsistencyRetrySession {
             // Connection to the contacted node is overloaded, try another one
             QueryError::UnableToAllocStreamId => RetryDecision::RetryNextNode(None),
             // In all other cases propagate the error to the user
-            _ => RetryDecision::DontRetry,
+            QueryError::BadQuery(_)
+            | QueryError::MetadataError(_)
+            | QueryError::CqlRequestSerialization(_)
+            | QueryError::BodyExtensionsParseError(_)
+            | QueryError::EmptyPlan
+            | QueryError::CqlResultParseError(_)
+            | QueryError::CqlErrorParseError(_)
+            | QueryError::ProtocolError(_)
+            | QueryError::TimeoutError
```

Comment:
Should we really propagate `TimeoutError` instead of trying another node? Well, it's the weird error about `use_keyspace`...

Reply:
Funny thing is that this error variant can't even appear there. AFAIU, `use keyspace` requests go through different path than regular requests. They are not a subject to retries, since we send `use keyspace` requests to all nodes simultaneously.